lib/pry/color.rb

```diff
@@ -0,0 +1,111 @@
+module Pry::Color
+  extend self
+  COLORS = {
+    "black"   => 0,
+    "red"     => 1,
+    "green"   => 2,
+    "yellow"  => 3,
+    "blue"    => 4,
+    "purple"  => 5,
+    "magenta" => 5,
+    "cyan"    => 6,
+    "white"   => 7
+  }
+  BRUSH_METHODS = [*COLORS.keys, *["bold", "default", "bright_default", "strip_color"]]
+  UnknownBrushError = Class.new(RuntimeError)
+
+  COLORS.each_pair do |color, value|
+    define_method color do |text, pry=((defined?(_pry_) and _pry_) or Pry)|
+      pry.color ? "\033[0;#{30+value}m#{text}\033[0m" : text
+    end
+
+    define_method "bright_#{color}" do |text, pry=((defined?(_pry_) and _pry_) or Pry)|
+      pry.color ? "\033[1;#{30+value}m#{text}\033[0m" : text
+    end
+    BRUSH_METHODS.push "bright_#{color}"
+
+    COLORS.each_pair do |bg_color, bg_value|
+      define_method "#{color}_on_#{bg_color}" do |text, pry=((defined?(_pry_) and _pry_) or Pry)|
+        pry.color ? "\033[0;#{30 + value};#{40 + bg_value}m#{text}\033[0m" : text
+      end
+
+      define_method "bright_#{color}_on_#{bg_color}" do |text, pry=((defined?(_pry_) and _pry_) or Pry)|
+        pry.color ? "\033[1;#{30 + value};#{40 + bg_value}m#{text}\033[0m" : text
+      end
+      BRUSH_METHODS.concat ["#{color}_on_#{bg_color}", "bright_#{color}_on_#{bg_color}"]
+    end
+  end
+
+  #
+  #
+  # @example
+  #   paint "foo", :bold
+  #
+  # @param [String] str
+  #   The string to paint.
+  #
+  # @param [String, Symbol] brush
+  #   The brush to paint 'str' with.
+  #
+  def paint str, brush, pry=((defined?(_pry_) and _pry_) or Pry)
+    if BRUSH_METHODS.include?(brush.to_s)
+      public_send brush, str, pry
+    else
+      raise UnknownBrushError, "#{brush} is not a known brush"
+    end
+  end
+
+  #
+  # @example
+  #   Pry::Color["hello", :green]
```

Comment:
This syntax is so weird, I've never seen anything like it in my life before. Where is it from?

Reply:
I remembered it from from the "Paint" Rubygem: https://github.com/janlelis/paint